python-sdk/src/astro/databricks/api_utils.py

```diff
@@ -0,0 +1,135 @@
+import fileinput
+import logging
+import os
+import pathlib
+import time
+from pathlib import Path
+
+from airflow.exceptions import AirflowException
+from databricks_cli.dbfs.api import DbfsApi, DbfsPath
+from databricks_cli.jobs.api import JobsApi
+from databricks_cli.runs.api import RunsApi
+from databricks_cli.sdk.api_client import ApiClient
+
+from astro.databricks.load_file.load_file_python_code_generator import render
+from astro.databricks.load_options import DeltaLoadOptions
+
+cwd = pathlib.Path(__file__).parent
+log = logging.getLogger(__name__)
+
+
+def generate_file(
+    data_source_path: str,
+    table_name: str,
+    source_type: str,
+    output_file_path: Path,
+    load_options: DeltaLoadOptions,
+    file_type: str = "",
+):
+    """
+    In order to run autoloader jobs in databricks, we need to generate a python file that creates a pyspark job.
+    This function uses jinja templating to generate a file with all necessary user inputs
+
+    :param data_source_path: The path where the data we're injesting lives (e.g. the s3 bucket URL)
+    :param table_name: The delta table we would like to push the data into
+    :param source_type: the source location (e.g. s3)
+    :param load_options: any additional load options the user would like to inject into their job
+    :param output_file_path: where the generated file should be placed.
+    :param file_type: when using the COPY INTO command, Spark requires explicitly stating the data type you are loading.
+        For individual files we can infer the file type, but if you are loading a directory please explicitly state
+        the file type in the File object.
+    :return: output file path
+    """
+    render(
+        Path("jinja_templates/load_file_to_delta.py.jinja2"),
+        {
+            "source_type": source_type,
+            "table_name": table_name,
+            "load_options": load_options,
+            "input_path": data_source_path,
+            "file_type": file_type.upper(),
+        },
+        output_file_path,
+    )
+
+    # Since we had to use jinja2 safe strings, we want to inline replace \' with '
+    # This is not a security concern now that jinja2 is no longer running.
+    for line in fileinput.input(output_file_path, inplace=True):
+        print(line.replace("\\'", "'"), end="")
```

Comment:
Do we want print here? or log?

Reply:
@kaxil this is not printing anything. It's just how fileinput works. this is how we replace the lines in-place.